### ISSUE TITLE
Destroy method 

### ISSUE THREAD

**Issue by davidpike7** [algolia/instantsearch.js#881]
Is there currently a way to destroy and then re-instantiate instant search?

I'm trying switch the index of the instant search instance but can't get it work

**Comment by vvo** [comment]
Hello, there's currently no destroy method, we might create one and this issue would be the start of it.

About your need: do you just want to keep all the widgets but change the index? If so you could do it with a custom widget. Let me know so that I can guide you doing so.

**Comment by davidpike7** [comment]
Hello again

We're be using (ideally) one search input with 4 different search indices and different widgets per index.

I was planning to instantiate and then destroy the instant instance when the user changed tab but ideally we wouldn't do that.

<img width="875" alt="screen shot 2016-03-01 at 21 46 46" src="https://cloud.githubusercontent.com/assets/1872727/13424866/60983630-dff7-11e5-9c8a-1e4bffcc92b5.png">

**Comment by vvo** [comment]
So yes to do so maybe the best option would be to:
1. create one instantsearch per tab
2. each instantsearch has his own searchbox
3. show/hide the corresponding searchbox based on the selected tab
4. show/hide the corresponding widgets based on the selected tab (from what I see, the widgets will be different: no price slider for news)
5. create a [custom widget](https://community.algolia.com/instantsearch.js/documentation/#custom-widgets) that you add to every instance of instantsearch and that will update the counts next to each tab
6. create a custom widget that you add to every instance of instantsearch and that will update the query (helper.setQuery) every time the widget goes from hidden/shown then redo a search to update the interface (helper.search())
6. on tab change, show the right widgets, update the query (instance.helper.setQuery)

This is rather complex but what you are trying to do is very advanced and requires this coding. Let me know what you think

**Comment by vvo** [comment]
As a quick solution, just doing a new pageload on tab change is also feasible, since the whole interface is updating.

**Comment by davidpike7** [comment]
The first approach has worked quite well.

I've gone about implementing it by listening for the helper `result` event and then using `setQuery` to  change the query for the other instances. I've used the stats widget to output the count for each instance. Obviously it triggers a lot of queries/renders though so not sure how performant it will be.

The only outstanding issue is that the searchbox is not updating when the query changes. Is there a way to trigger a re-render or manually set it?

**Comment by vvo** [comment]
Awesome! Do you feel like writing a tutorial that we would publish on our [blog](https://blog.algolia.com/)? This seems to be like one of the most advanced usages of instantsearch.js that we did. Contact me at vincent@algolia.com if interested.

> Obviously it triggers a lot of queries/renders though so not sure how performant it will be.

Yes indeed that's why I proposed a solution where only the count is updated. We might have to re-think about this usecase once we have more feedback from you to see what could be done.


> The only outstanding issue is that the searchbox is not updating when the query changes. Is there a way to trigger a re-render or manually set it?

Yes this is a known issue that I fixed yesterday, should be released soon (see #886)

In the meantime you can update the value of the input manually with DOM:

```js
document.querySelector('#search-box input').value = ..;
helper.setQuery(..).search();
```

Does that makes sense?

**Comment by davidpike7** [comment]
I'd be happy to. The code is still a bit rough and we've still got a way to go with the formatting of the content but will get in touch once we've progressed it a bit further.

Cheers for the heads up on the search box.

How would you recommend creating a widget that just handles the count? I was thinking of using the 'result` event to trigger `AlgoliaSearchHelper.searchOnce` for the other indices.

**Comment by vvo** [comment]
> How would you recommend creating a widget that just handles the count? I was thinking of using the result event to trigger AlgoliaSearchHelper.searchOnce for the other indices.

Yes that was the point of my

> 5. create a custom widget that you add to every instance of instantsearch and that will update the counts next to each tab

which was not enough detailed but you could create this widget and use the render() method to get an event when the search was really updated for the underlying instantsearch instance.

Then call otherInstances.helper.searchOnce(callback) and extract in the callback the number of hits for the hidden widgets.

Is that more clear?

**Comment by vvo** [comment]
destroy method will be implemented via #1065
